MetaMask/metamask-extension
# Handling custom token decimal fetch failure due to network error

Fixes: #9777 #10534 #11081

Explanation:  Currently while fetching the custom token if the decimal value fetch fails, it is defaulting to 0 without warning the user.

Fix:- If decimal value fetch fails the field will become editable and shows warning messages for network failure.

Manual testing steps:

- Create a custom token from test-dapp, and copy the token address.
- From the extension homepage, click the `Add Token` button, in the `Custom Token` tab and enter the token address.
  - The default behavior should be that the symbol and decimal will get populated automatically and the `Next` button will be active.
  - Based on the current changes when the decimal fetch fails, the decimal tab will be empty and editable, a warning message will be shown below the text field with a link to verify the token decimal and the `Next` button will be disabled.

https://user-images.githubusercontent.com/43930900/116836899-5ede4080-ab96-11eb-9e08-095147eae58f.mov
